### PR TITLE
Add Document Creation Function

### PR DESCRIPTION
**Pull request by en3on**

### CODE REVIEW

File: controllers/documents-controller.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const {verifyToken} = require('../utils/auth-utils.js');
+const mongoose = require('mongoose');
+
+const uploadFile = require('../utils/file-uploader.js');
+
+async function uploadHandler(req, res, next) {
+  const {token} = req.headers;
+  const {files} = req;
+  const {descriptions} = req.body;
+  // console.log({token, descriptions, files});
```
Comment (PraveenMut) [inline controllers/documents-controller.js:12]: strangler console.log spotted -- not sure if you want to keep it or not.

File: controllers/documents-controller.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  try {
+    let user = verifyToken(token);
+
+    user = await User.findOne({email: user});
+
+    // create document and assign it to user
+    for (i = 0; i < files.length; i++) {
+      const resp = await uploadFile(files[i].buffer);
```
Comment (PraveenMut) [inline controllers/documents-controller.js:21]: just a quick question, the .buffer allows you to read the file right?

File: test/utils/post-request-helpers.js [hunk 3]
```diff
@@ -1,12 +1,33 @@
...
+
   return await request(api)
-    .post(url)
-    .send(payload)
-    .set('Accept', 'application/json');
+      .post(url)
+      .field('descriptions', ['file 1', 'file 2'])
+      .attach('documents', 'test/utils/files/1.txt')
+      .attach('documents', 'test/utils/files/2.txt')
+      .set(headers);
```
Comment (PraveenMut) [inline test/utils/post-request-helpers.js:34]: wow this is amazing, need to learn how you did this, maybe you can explain later haha

**Review by PraveenMut — APPROVED** [review]
Good stuff, I was trying to figure out how you created those sweet unit tests. Also, looking at this, so sending off an array of file objects should be okay then. We can discuss when we meet.

**Review by alex-double-u-old — APPROVED** [review]
Ladell = GOD

**Comment by PraveenMut** [comment]
> Ladell = GOD

const GOD = "ladell" lol

**Pull request merged**